Storage/Processor/Processor.php

```diff
@@ -271,42 +271,26 @@ private function getResponseFromStreamInterface(StreamInterface $streamInterface
             $response->headers->set('Accept-Ranges', $request->isMethodSafe() ? 'bytes' : 'none');
         }
 
-        $range = $request->headers->get('Range');
-
-        if ($range === null) {
-            return $response;
-        }
-
-        list($start, $end) = explode('-', substr($range, 6), 2) + [0];
-
-        $end = ('' === $end) ? $fileSize - 1 : (int) $end;
-
-        if ('' === $start) {
-            $start = $fileSize - $end;
-            $end = $fileSize - 1;
-        } else {
-            $start = (int) $start;
-        }
-
-        if ($start > $end) {
+        $streamRange = new StreamRange($request->headers, $fileSize);
+        if ($streamRange->isOutOfRange()) {
             return $response;
         }
 
-        if ($start < 0 || $end > $fileSize - 1) {
+        if (!$streamRange->isSatisfiable()) {
```

Comment:
Why is there a difference between "isOutOfRange" and "isSatisfiable"? Aren't both checking "out of range" using 3 different checks?

Reply:
It's based to the Symfony BinaryResponse:

- [in case of out-of-range it's just retuning the binaries from byte 0 with a return code 200](https://github.com/symfony/http-foundation/blob/master/BinaryFileResponse.php#L250)
- i[n case of not satifiable it's also returning from byte 0 but with the return code HTTP_REQUESTED_RANGE_NOT_SATISFIABLE
](https://github.com/symfony/http-foundation/blob/master/BinaryFileResponse.php#L250)

So it's not the same